### PR TITLE
Video email in html format

### PR DESCRIPTION
**Pull request by gnott**
There were some video emails coming from the workflows today, good! The bad is the HTML content was all in plain text. 

Here is a little tweak to support `text/plain` and `text/html` emails sent from the email provider. I also added in the `charset` option. If the `charset` is `utf-8` then the email body is base64 encoded. I don't know if we always want that, so there are some options to leave the charset as `None` for sending really simple emails.

Hopefully sending in multipart emails using this provider will work for all the current situations. If it can send simple plain messages, HTML emails, and emails with attachments, I think it will cover all the uses we have.

### CODE REVIEW

File: tests/provider/test_email_provider.py [hunk 1]
```diff
@@ -58,3 +59,32 @@ def test_valid_valid_recipient_object(self, e_mail, expected):
...
+        # note: boundary value is not constant so cannot compare with a fixture easily
+        expected_fragments.append('Content-Type: multipart/mixed; boundary=')
+        expected_fragments.append('Subject: %s' % subject)
+        expected_fragments.append('From: %s' % sender)
+        expected_fragments.append('To: %s' % recipient)
+        expected_fragments.append(expected_content_type)
+        expected_fragments.append('MIME-Version: 1.0')
+        expected_fragments.append('Content-Transfer-Encoding: base64')
+        # body is base64 encoded
+        expected_fragments.append(base64_encode_string(body))
```
Comment (giorgiosironi) [inline tests/provider/test_email_provider.py]: not perfect for readability (of the source), but I very seldom open an email original source

Comment (gnott) [reply tests/provider/test_email_provider.py]: It was funny when I viewed the email as a string to see it. I think it only happens when the charset is `utf-8` (and I don't know if it is controllable in the code libraries). I tried to make the test case as readable as possible.

**Comment by gnott** [comment]
Thanks for the comments so far, with the current usage of the email provider as all internal recipients, it should be fairly low risk to failure, and I'm really curious to see the result of the new email headers, so I will merge now.

File: tests/provider/test_email_provider.py (hunk 1 above)
Comment (gnott) [reply tests/provider/test_email_provider.py]: As part of bug fixing emails, I have a function in `tests/activity/test_activity_validate_digest_input.py` that will extract the body from a multipart email and decode the body. It's a little rough but it could be expanded to other tests when required.
